### PR TITLE
[RAPTOR-10100] bump pyarrow<=14.0.1

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Bump pyarrow due to CVE
https://nvd.nist.gov/vuln/detail/CVE-2023-47248

Also update missed pillow dep

### CODE REVIEW

**Review by c-h-russell-walker — APPROVED** [review]
LGTM - thank you for the update!

**Comment by yakov-g** [comment]
jenkins run test_integration_r_environment

**Comment by yakov-g** [comment]
jenkins run test_training_model_templates

File: example_dropin_environments/julia_mlj/dr_requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 pandas==1.0.5
-pyarrow==0.14.1,<=10.0.1
+pyarrow==0.14.1,<=14.0.1
```
Comment (c-h-russell-walker) [inline example_dropin_environments/julia_mlj/dr_requirements.txt:2]: @yakov-g I just found that our trviy checks failed on pyarrow still 
https://jenkins2.devinfra.drdev.io/job/CodeExperience-addons/job/notebooks-jenkins_trivy_groovy/257/artifact/reports/trivy_kernel-python-3.9-gpu.html

I apologize for not taking a closer look at this PR (I am _relatively_ unfamiliar with this repo.)
but it looks like we need to update multiple other places ... ??

https://github.com/search?q=repo%3Adatarobot%2Fdatarobot-user-models+pyarrow%3E%3D0.14.1%2C%3C%3D10.0.1&type=code

Screenshot 2023-11-15 at 9 33 38 AM

**Pull request merged**